extensions/wikia/UserProfilePageV3/js/UserProfilePage.js

```diff
@@ -1,94 +1,100 @@
-				$(event.target).attr('data-confirm'));
-		});
-
-		$('#userAvatarEdit').click(function (event) {
-			event.preventDefault();
-			UserProfilePage.renderLightbox('avatar');
-		});
-
-		$('#discussionAllPostsByUser').click(function (event) {
-			Wikia.Tracker.track({
-				action: Wikia.Tracker.ACTIONS.CLICK_LINK_TEXT,
-				browserEvent: event,
-				href: $(event.target).attr('href'),
-				label: 'discussion-all-posts-by-user'
+			$userIdentityBoxEdit.click(function (event) {
+				event.preventDefault();
+				UserProfilePage.renderLightbox('about');
 			});
-		});
 
-		// for touch devices (without hover state) make sure that Edit is always
-		// visible
-		if (Wikia.isTouchScreen()) {
-			$userIdentityBoxEdit.show();
-		}
-	},
+			$('#UserAvatarRemove').click(function (event) {
+				event.preventDefault();
+				UserProfilePage.removeAvatar($(event.target).attr('data-name'),
+					$(event.target).attr('data-confirm'));
+			});
 
-	renderLightbox: function (tabName) {
-		'use strict';
+			$('#userAvatarEdit').click(function (event) {
+				event.preventDefault();
+				UserProfilePage.renderLightbox('avatar');
+			});
 
-		if (!UserProfilePage.isLightboxGenerating) {
-			// Start profiling
-			UserProfilePage.bucky.timer.start('renderLightboxSuccess');
-			UserProfilePage.bucky.timer.start('renderLightboxFail');
+			$('#discussionAllPostsByUser').click(function (event) {
+				tracker.track({
+					action: tracker.ACTIONS.CLICK_LINK_TEXT,
+					browserEvent: event,
+					href: $(event.target).attr('href'),
+					label: 'discussion-all-posts-by-user'
+				});
+			});
 
-			//if lightbox is generating we don't want to let user open second one
-			UserProfilePage.isLightboxGenerating = true;
-			UserProfilePage.newAvatar = false;
+			$('#userIdentityBoxClear').click(function (event) {
+				event.preventDefault();
+				UserProfilePage.clearMastheadContents(
+					$(event.target).attr('data-name'),
+					$(event.target).attr('data-confirm')
+				);
+			});
+
+			// for touch devices (without hover state) make sure that Edit is always
+			// visible
+			if (window.Wikia.isTouchScreen()) {
+				$userIdentityBoxEdit.show();
+			}
+		},
+
+		renderLightbox: function (tabName) {
+			if (!UserProfilePage.isLightboxGenerating) {
+				// Start profiling
+				UserProfilePage.bucky.timer.start('renderLightboxSuccess');
+				UserProfilePage.bucky.timer.start('renderLightboxFail');
+
+				//if lightbox is generating we don't want to let user open second one
+				UserProfilePage.isLightboxGenerating = true;
+				UserProfilePage.newAvatar = false;
+
+				$.when(
+					// TODO: this should probably be refactored as mustache template
+					nirvana.sendRequest({
+						controller: 'UserProfilePage',
+						method: 'getLightboxData',
+						type: 'GET',
+						data: {
+							tab: tabName,
+							userId: UserProfilePage.userId,
+							rand: Math.floor(Math.random() * 100001) // is cache buster really needed here?
```

Comment:
Let's get rid of `rand` parameter here and simply set proper headers to avoid caching of the response.